crates/starknet-devnet-server/src/api/error.rs

```diff
@@ -83,12 +82,23 @@ pub enum ApiError {
     ContractClassSizeIsTooLarge,
     #[error("Minting reverted")]
     MintingReverted { tx_hash: Felt, revert_reason: Option<String> },
+    #[error("The dump operation failed: {msg}")]
+    DumpError { msg: String },
+    #[error("Messaging error: {msg}")]
+    MessagingError { msg: String },
+    #[error("Invalid value: {msg}")]
+    InvalidValueError { msg: String },
```

Comment:
I know that this PR just moved them from the deleted `HttpApiError` enum, but `GeneralError` and `InvalidValueError` look too generic to me. `InvalidValueError` is only returned for failed address parsing, which in most cases arises from `get_erc20_address`, which could even be made not to return a `Result`. `GeneralError` is only used for failed `get_balance`. I can move this to a separate issue.